### PR TITLE
SNOW-1644950: make use_logical_type option more explicit

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1644950

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Move `use_logical_type` option outside of `kwargs` and make it more explicit. All existing tests should pass.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 #### Improvements
 
-- Added support for `ln` in `snowflake.snowpark.functions`
+- Improved documentation for `Session.write_pandas` by making `use_logical_type` option more explicit.
```
Comment (sfc-gh-aalam) [inline CHANGELOG.md:15]: removed `ln` function since it is already added above.

File: src/snowflake/snowpark/session.py [hunk 2]
```diff
@@ -2429,6 +2430,11 @@ def write_pandas(
             table_type: The table type of table to be created. The supported values are: ``temp``, ``temporary``,
                 and ``transient``. An empty string means to create a permanent table. Learn more about table types
                 `here <https://docs.snowflake.com/en/user-guide/tables-temp-transient.html>`_.
+            use_logical_type: Boolean that specifies whether to use Parquet logical types when reading the parquet files
+                for the unloaded pandas dataframe. With this file format option, Snowflake can interpret Parquet logical
```
Comment (sfc-gh-jrose) [inline src/snowflake/snowpark/session.py:2434]:
```suggestion
                for the uploaded pandas dataframe. With this file format option, Snowflake can interpret Parquet logical
```

**Review by sfc-gh-jrose — APPROVED** [review]

File: src/snowflake/snowpark/session.py [hunk 3]
```diff
@@ -2429,6 +2430,11 @@ def write_pandas(
             table_type: The table type of table to be created. The supported values are: ``temp``, ``temporary``,
                 and ``transient``. An empty string means to create a permanent table. Learn more about table types
                 `here <https://docs.snowflake.com/en/user-guide/tables-temp-transient.html>`_.
+            use_logical_type: Boolean that specifies whether to use Parquet logical types when reading the parquet files
+                for the uploaded pandas dataframe. With this file format option, Snowflake can interpret Parquet logical
+                types during data loading. To enable Parquet logical types, set use_logical_type as True. Set to None to
+                use Snowflakes default. For more information, see:
+                `file format options: <https://docs.snowflake.com/en/sql-reference/sql/create-file-format#type-parquet>`_.
```
Comment (sfc-gh-aling) [inline src/snowflake/snowpark/session.py:2437]: is it only supported starting from a certain version of connector?
should we log warning if use_logical_type not in signature.parameters but still specified by user in this case?

Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/session.py:2437]: we do that already

Comment (sfc-gh-aling) [reply src/snowflake/snowpark/session.py:2437]: yep, this is what I was asking, thanks for pointing

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**